src/Controllers/cli_controller.py

```diff
@@ -1,22 +1,26 @@
-
 from Models.uml_diagram import UML_Diagram
-
 from prompt_toolkit import prompt
 from Views.cli_view import CLI_View
+from prompt_toolkit.completion import NestedCompleter
+
 
 class CLI_Controller:
 
     def __init__(self):
         self._view = CLI_View()
-    
+
     def request_update(self):
-         return prompt("Command: ", completer=self._view._completer).strip()
+        self.setup_autocomplete()
+        return prompt("Command: ", completer=self._completer).strip()
 
     def draw(self, diagram: UML_Diagram):
         pass
 
-#=========================CLI Specific Parseing=========================#  
-    def parse_list_cmd(self, d:UML_Diagram, tokens:list[str]):
+    def get_diagram(self, diagram: UML_Diagram):
```

Comment:
basically, delete this method. It isn't a getter, it violates python convention, it represents poor program design.